vertx-web/src/main/java/io/vertx/ext/web/RoutingContext.java

```diff
@@ -613,7 +613,19 @@ default LanguageHeader preferredLanguage() {
   }
 
   /**
-   * Returns a map of named parameters as defined in path declaration with their actual values
+   * Add a new one or replace an existing path parameter
+   * @throws NullPointerException when the name or value is null
+   */
+  void addOrReplacePathParam(String name, String value);
```

Comment:
Do we really need this new method? It will create the underlying map if needed anyway.

Reply:
good point; it was consistent to not expose the underlying form/nature/content of the map - in order to enable to not allocate it, if possible

But any suggestion is more than welcome, bud :pray:  and thanks for looking at this!